File: day1/main.exs
res, s= File.read("input.txt")
ar = String.split(s,"\n")

ar |> List.delete_at(-1) |> Enum.map(fn x-> x|> String.to_charlist() |> Enum.filter(fn x-> x <65 end) |>(&[Enum.at(&1, 0), Enum.at(&1, -1)]).() |> List.to_integer() end) |> Enum.reduce( 0, &+/2)

File: day1/main.rs
use std::fs;

fn main() {

  let contents = fs::read_to_string("input.txt")
      .expect("Should have been able to read the file");
  let x = contents.split('\n');
  let mut sum: i128 = 0;
  
  for str in x{
    let arr: Vec<char> = str.chars().filter(|f| f.is_numeric()).collect();
    let mut s= String::from(arr.first().unwrap().to_string() + & arr.last().unwrap().to_string());

    let z: i128 =s.parse().unwrap();
    sum += z;  
  }
   println!("{sum}");
}
